rfcs/text/0000-extend-file-pe.md

```diff
@@ -0,0 +1,142 @@
+# 0000: Extend the PE field set
+
+- Stage: **1 (proposal)**
+- Date: **TBD**
+
+The Portable Executable (PE) sub-field, of the `file` top-level fieldset, can be updated to include more file attributes to aid in file analysis. This additional document metadata can be used for malware research, as well as coding and other application development efforts.
+
+## Fields
+
+This RFC is to create 25 additional sub-fields within the `file.pe` fieldset.
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| file.pe.authentihash | keyword | Authentihash of the PE file. |
+| file.pe.compile_timestamp | date | Compile timestamp of the PE file. |
+| file.pe.compiler | keyword | Compiler information. |
+| file.pe.compiler.version | keyword | Version of the compiler. |
+| file.pe.compiler.name | keyword | Name of the compiler. |
+| file.pe.creation_date | date | Extracted when possible from the file's metadata. Indicates when it was built or compiled. It can also be faked by malware creators. |
+| file.pe.entry_point | keyword | Entry point of the PE file. |
+| file.pe.exports | keyword | List of symbols exported by PE |
+| file.pe.debug | nested | Debug information, if present |
+| file.pe.debug.offset | keyword | Debug offset information. |
+| file.pe.debug.size | keyword | Size of the debug information. |
+| file.pe.debug.type_str | keyword | Format of the debug information. |
+| file.pe.debug.type | keyword | Information type generated by the debug options. |
+| file.pe.debug.timestamp | date | Timestamp of the debug information. |
+| file.pe.imports | flattened | List of all imported functions |
+| file.pe.sections | nested | Data about sections of compiled binary PE |
+| file.pe.sections.chi2 | long | Chi-square probability distribution. |
+| file.pe.sections.virtual_address | long | Virtual address available to the file. |
+| file.pe.sections.entropy | float | Measurement of entropy randomness in the file. |
+| file.pe.sections.flags | keyword | Section flags of the file. |
+| file.pe.sections.name | keyword | Section names of the file. |
+| file.pe.sections.raw_size | long | Size of the section or the dize of the initialized data on disk. |
+| file.pe.resources | nested | If the PE contains resources, some info about them |
```

Comment:
This looks like it needs to be updated with the schema of the fields in the nested document (i.e. `file.pe.resources.language`, etc.)